--- a/open-vm-tools/lib/syncDriver/syncDriverLinux.c
+++ b/open-vm-tools/lib/syncDriver/syncDriverLinux.c
@@ -34,6 +34,7 @@
 #include <sys/stat.h>
 #include "debug.h"
 #include "dynbuf.h"
+#include "fileIO.h"
 #include "syncDriverInt.h"
 
 /* Out toolchain headers are somewhat outdated and don't define these. */
@@ -170,6 +171,9 @@
    DynBuf fds;
    LinuxDriver *sync = NULL;
    SyncDriverErr err = SD_SUCCESS;
+   FileIOResult status;
+   FileIODescriptor tmpFD;
+   const char *touchPath = "/run/pexip/vm_frozen";
 
    DynBuf_Init(&fds);
 
@@ -189,6 +193,12 @@
     */
    VERIFY(paths != NULL);
 
+   FileIO_Invalidate(&tmpFD);
+   status = FileIO_Open(&tmpFD, touchPath, FILEIO_OPEN_ACCESS_WRITE, FILEIO_OPEN_CREATE_EMPTY);
+   if (FileIO_IsSuccess(status)) {
+      FileIO_Close(&tmpFD);
+   }
+
    /*
     * Iterate through the requested paths. If we get an error for the first
     * path, and it's not EPERM, assume that the ioctls are not available in
