--- a/open-vm-tools/configure.ac
+++ b/open-vm-tools/configure.ac
@@ -1896,7 +1896,6 @@
    services/plugins/powerOps/Makefile  \
    services/plugins/resolutionSet/Makefile \
    services/plugins/resolutionKMS/Makefile \
-   services/plugins/timeSync/Makefile  \
    services/plugins/vix/Makefile       \
    services/plugins/vmbackup/Makefile  \
    services/plugins/deployPkg/Makefile  \
--- a/open-vm-tools/services/plugins/Makefile.am
+++ b/open-vm-tools/services/plugins/Makefile.am
@@ -51,7 +51,6 @@
 endif
 SUBDIRS += resolutionKMS
 endif
-SUBDIRS += timeSync
 SUBDIRS += vix
 SUBDIRS += vmbackup
 
